plugins/inputs/jboss/jboss.go

```diff
@@ -0,0 +1,1312 @@
+	for key, value := range server.Result {
+		switch key {
+		case "bytesReceived", "bytesSent", "requestCount", "errorCount", "maxTime", "processingTime":
+			if value != nil {
+				switch value.(type) {
+				case int:
+					fields[key] = value.(float64)
+				case float64:
+					fields[key] = value.(float64)
+				case string:
+					f, err := strconv.ParseFloat(value.(string), 64)
+					if err != nil {
+						log.Printf("E! JBoss Error decoding Float  from string : %s = %s\n", key, value.(string))
+					} else {
+						fields[key] = f
+					}
+				}
+			}
+		}
+	}
+	tags := map[string]string{
+		"jboss_host":   host,
+		"jboss_server": serverName,
+		"type":         connector,
+	}
+	acc.AddFields("jboss_web", fields, tags)
+
+	return nil
+}
+
+func (h *JBoss) getUndertowStatistics(
+	acc telegraf.Accumulator,
+	serverURL string,
+	execAsDomain bool,
+	host string,
+	serverName string,
+	listener string,
+) error {
+	adr := OrderedMap{}
+	var listenerName = "default"
+	if listener == "ajp" || listener == "https" {
+		listenerName = listener
+	}
+	listener = listener + "-listener"
+	if execAsDomain {
+		adr = OrderedMap{
+			{"host", host},
+			{"server", serverName},
+			{"subsystem", "undertow"},
+			{"server", "default-server"},
+			{listener, listenerName},
+		}
+	} else {
+		adr = OrderedMap{
+			{"subsystem", "undertow"},
+			{"server", "default-server"},
+			{listener, listenerName},
+		}
+	}
+
+	bodyContent, err := h.createRequestBody(GetWebStat, adr)
+	if err != nil {
+		return fmt.Errorf("error on request to %s : %s\n", serverURL, err)
+	}
+
+	out, err := h.doRequest(serverURL, bodyContent)
+
+	if err != nil {
+		return fmt.Errorf("error on request to %s OUT: %s  ERR: %s\n", serverURL, out, err)
+	}
+	server := WebResponse{}
+	if err = json.Unmarshal(out, &server); err != nil {
+		return fmt.Errorf("Error decoding JSON response: OUT: %s  ERR: %s", out, err)
+	}
+
+	fields := make(map[string]interface{})
+	for key, value := range server.Result {
+		log.Printf("D! LISTERNER %s : %s \n", key, value)
+		switch key {
+		case "bytes-received", "bytes-sent", "request-count", "error-count", "max-processing-time", "processing-time":
+			if value != nil {
+				switch value.(type) {
+				case int:
+					fields[key] = value.(float64)
+				case float64:
+					fields[key] = value.(float64)
+				case string:
+					f, err := strconv.ParseFloat(value.(string), 64)
+					if err != nil {
+						log.Printf("E! JBoss Error decoding Float  from string : %s = %s\n", key, value.(string))
+					} else {
+						fields[key] = f
+					}
+				}
+			}
+		}
+	}
+	tags := map[string]string{
+		"jboss_host":   host,
+		"jboss_server": serverName,
+		"type":         listener,
+	}
+	acc.AddFields("jboss_web", fields, tags)
+
+	return nil
+}
+
+func GetPoolFields(pool DBPoolStatistics) map[string]interface{} {
+	retmap := make(map[string]interface{})
+	//Jboss EAP 6/AS 7.X returns "strings", wildfly 12 returns integers
+	switch pool.ActiveCount.(type) {
```

Comment:
Please use type-switches more efficiently.
```suggestion
	switch v := pool.ActiveCount.(type) {
```